reporting/src/main/java/org/wso2/testgrid/reporting/EmailReportGenerator.java

```diff
@@ -0,0 +1,205 @@
+     * @return Summary of integration test results
+     */
+    private String getIntegrationTestLogResult(TestPlan testPlan) throws ReportingException {
+        StringBuilder stringBuilder = new StringBuilder();
+        Path filePath;
+        try {
+            filePath = Paths.get(TestGridUtil.deriveTestIntegrationLogFilePath(testPlan, false));
+        } catch (TestGridException e) {
+            throw new ReportingException("Error occurred while deriving integration-test log file path to get" +
+                    " integration test results for test-plan with id " + testPlan.getId() + "of the product " +
+                    testPlan.getDeploymentPattern().getProduct().getName() + "having infra-combination as " +
+                    testPlan.getInfraParameters(), e);
+        }
+
+        if (Files.exists(filePath)) {
+            try (BufferedReader bufferedReader = Files.newBufferedReader(filePath)) {
+                String currentLine;
+                int lineCount = 0;
+                while ((currentLine = bufferedReader.readLine()) != null) {
+                    stringBuilder.append(currentLine).append(HTML_LINE_SEPARATOR);
+                    if (++lineCount == 100) {
+                        stringBuilder.append(".........").append(HTML_LINE_SEPARATOR).append(".........")
+                                .append(HTML_LINE_SEPARATOR).append("(view complete log in detailed results..)");
+                        break;
+                    }
+                }
+                return stringBuilder.toString();
+            } catch (IOException e) {
+                throw new ReportingException("Error occurred while reading integration-test log file to get" +
+                        " integration test results for test-plan with id " + testPlan.getId() + "of the product " +
+                        testPlan.getDeploymentPattern().getProduct().getName() + "having infra-combination as " +
+                        testPlan.getInfraParameters(), e);
+            }
+        } else {
+            if (testPlan.getStatus().equals(Status.SUCCESS) || testPlan.getStatus().equals(Status.FAIL)) {
+                logger.error("Integration-test log file does not exists for test-plan with id "
+                        + testPlan.getId() + "of the product " +
+                        testPlan.getDeploymentPattern().getProduct().getName() + "having infra-combination as " +
+                        testPlan.getInfraParameters());
+                return "Integration test-log is missing. Please contact TestGrid administrator.";
+            } else {
+                return "Test status is " + testPlan.getStatus() + ". Hence can not display the log.";
+            }
+        }
+    }
+
+    /**
+     * Returns the current status of the product.
+     * @param product product
+     * @return current status of the product
+     */
+    public Status getProductStatus(Product product) {
+        TestPlanUOW testPlanUOW = new TestPlanUOW();
```

Comment:
Instantiating UOW classes inside methods breaks the unit tests. Please move to constructor if you really need this at this level. See GenerateTestPlanCommand class for an example.